src/Wizard/Wizard.jsx

```diff
@@ -42,11 +42,19 @@ export class Wizard extends React.Component {
   }
 
   nextStepHandler() {
-    if (this.state.currentStep === this.props.steps.length - 1) {
-      this.props.onComplete(this.state.data);
+    const {steps, onComplete} = this.props;
+    const {currentStep, data} = this.state;
+
+    // If an onComplete handler is provided for the current step, and it returns false, we won't
+    // proceed to the next step.
+    if (steps[currentStep].onComplete && !steps[currentStep].onComplete(data)) {
```

Comment:
Down the road, it'd be great if we built this error handling into the component, but we haven't yet ported the `ToastNotification` into Clever components, which is our primary method of surfacing errors (at least in the Apps Dashboard).

For now, I think that it's okay to keep this general. As a result, the values provided to `onStepComplete` will look very similar to the `onChange` handlers provided to buttons, etc. (here's an [example](https://github.com/Clever/apps-dashboard/blob/bc30edf1e78401cd3a56d6c880fb71e6e67475fe/pages/partner/applications/ui/components/panels/UpdateFooter.jsx#L31-L55)). The `onStepComplete` values will just have added `return [true|false]` statements.

Happy to talk through this in person if you have additional thoughts!